### ISSUE TITLE
[Feature]: API to get current form values by formId

### ISSUE THREAD

**Issue by danecando** [airjp73/remix-validated-form#118]
### What is the new or updated feature that you are suggesting?

An API to get the current input values from the form state. 

### Why should this feature be included?

I didn't notice any way in the current API to get or inspect the values from the form. Would be useful for quick debugging and features like having a preview component that gets updated while a form is being edited.

**Comment by jbstewart** [comment]
I tried the suggestion of using a formRef, but Remix gives me an error about including Server code on the Client when I try it. Does anyone have an example of using formRef to retrieve the values of a form on the client they could point me to?

**Comment by airjp73** [comment]
> I tried the suggestion of using a formRef, but Remix gives me an error about including Server code on the Client when I try it. Does anyone have an example of using formRef to retrieve the values of a form on the client they could point me to?

This is most likely unrelated. Sometimes remix isn't able to figure out which code is client code and which is server code and something seemingly unrelated can trigger an issue with this, in my experience. There's a good [doc on this in the remix docs](https://remix.run/docs/en/v1/pages/gotchas#server-code-in-client-bundles).

**Comment by jbstewart** [comment]
Here's what I have. The line with `new FormData(ref)` is what causes Remix to throw that error.

<img width="937" alt="image" src="https://user-images.githubusercontent.com/2489288/172501571-6348533d-fa31-433f-9933-8fbbc770ee12.png">

**Comment by airjp73** [comment]
The issue is likely elsewhere in the file. Could you share the whole file?

**Comment by danecando** [comment]
You probably need to use `const formData = new FormData(ref.current)`

**Comment by airjp73** [comment]
Ah, good catch.

**Comment by jbstewart** [comment]
@danecando Yes, good catch - but unfortunately that didn't fix the problem. Here's a gist of the whole file:
<script src="https://gist.github.com/jbstewart/6198442bc5937642b3bb5dab662b56aa.js"></script>

**Comment by airjp73** [comment]
I'm pretty sure this is the issue.

```ts
import { FormData, json, redirect } from '@remix-run/node'
```

You don't need to import `FormData`, it's a global variable. In the browser it's built-in, but here you're importing and using the node version of `FormData` that remix provides. Remix also makes this globally available in the server code.

**Comment by jbstewart** [comment]
@airjp73 Thanks Aaron! That was it.

**Comment by airjp73** [comment]
Following up on this: I think the path forward on this one is to add a helper to `useFormContext` that does something like this:

```ts
getFormData: () => new FormData(formElement),
```

Going to tag this as a good first issue since this functionality is already achievable. Anyone feel free to reach out if you want to take a look at this and need some guidance.

**Comment by airjp73** [comment]
New helper published in remix-validated-form v4.5.1
